src/components/notification/channel/desktop.ts

```diff
@@ -22,12 +22,77 @@
  * SOFTWARE.                                                                      *
  **********************************************************************************/
 
-import { notify } from 'node-notifier'
+import { hostname, type } from 'os'
+import { spawnSync, execSync } from 'child_process'
 import { DesktopData } from './../../../interfaces/data'
 import getIp from '../../../utils/ip'
-import { hostname } from 'os'
 import { publicIpAddress } from '../../../utils/public-ip'
 
+interface NotifyData {
+  title: string
+  message: string
+}
+
+/**
+ * Escape a string for PowerShell.
+ * @param {string} str input string
+ * @return {string} escape single quote with another
+ */
+export const psEscape = (str: string) => {
+  let result = ''
+  for (let i = 0; i < str.length; i++) {
+    const ch = str[i]
+    if (ch.charCodeAt(0) === 39) {
+      // single quote, escape it with another single quote
+      result += ch
+    }
+    result += ch
+  }
+  return result
+}
+
+/**
+ * Send notification according to the operating system
+ * @param {NotifyData} data notification data
+ * @return {void}
+ */
+const notify = (data: NotifyData) => {
+  const { title, message } = data
+  const operatingSystem = type()
+
+  // OSAScript for OS X
+  const osascript = `display notification "${message}" with title "${title}"`
+
+  // Powershell Script for Windows
+  const powershellScript = `
+  [Windows.UI.Notifications.ToastNotificationManager, Windows.UI.Notifications, ContentType = WindowsRuntime] > $null;
+  $templateType = [Windows.UI.Notifications.ToastTemplateType]::ToastText02;
+  $template = [Windows.UI.Notifications.ToastNotificationManager]::GetTemplateContent($templateType);
+  $template.SelectSingleNode("//text[@id=1]").InnerText = '${psEscape(title)}';
+  $template.SelectSingleNode("//text[@id=2]").InnerText = '${psEscape(
+    message
+  )}';
+  $toast = [Windows.UI.Notifications.ToastNotification]::new($template);
+  $notifier = [Windows.UI.Notifications.ToastNotificationManager]::CreateToastNotifier('Monika');
+  $notifier.Show($toast);
+  `
+
+  switch (operatingSystem) {
+    case 'Linux':
+      spawnSync('notify-send', ['-a', 'Monika', title, message])
+      break
+    case 'Darwin':
+      spawnSync('osascript', ['-e', osascript])
```

Comment:
Running on MacOS

![Screen Shot 2021-08-09 at 14 57 15](https://user-images.githubusercontent.com/22452017/128675405-2fbcf37c-2983-484b-a9ad-1d642f4c0edf.png)


When one of those notification is clicked, this window opened

![Screen Shot 2021-08-09 at 14 57 42](https://user-images.githubusercontent.com/22452017/128675464-216b9c05-67df-4d05-993f-cd35b1486c8f.png)

Reply:
Is it possible to change notification card title from "SCRIPT EDITOR" to "Monika"?

Reply:
I updated with title and subtitle, please take a look. I don't have a Mac device so I couldn't test this.

Reply:
seems like it's not possible to change the "SCRIPT EDITOR" title. So maybe just remove the subtitle. It looks redundant. 
<img width="364" alt="Screenshot 2021-08-10 at 00 00 51" src="https://user-images.githubusercontent.com/311343/128780130-be6fbde0-7001-4c9b-a86a-0e42a1797d9f.png">

Reply:
We still need the subtitle though, 
![image](https://user-images.githubusercontent.com/16670475/128798160-1daa5f14-6d63-4aa1-a09a-360ccafc2bcd.png)

Title is for the app name, which is Monika
Subtitle is for the event (Monika is Running, New INCIDENT, new RECOVERY)
Message is for message

Reply:
I think we should use title for Monika, considering we couldn't remove the "Script Editor" title.